modin/core/execution/dask/implementations/pandas_on_dask/partitioning/partition_manager.py

```diff
@@ -48,3 +52,150 @@ def get_objects_from_partitions(cls, partitions):
             The objects wrapped by `partitions`.
         """
         return DaskWrapper.materialize([partition._data for partition in partitions])
+
+    @classmethod
+    def concat(cls, axis, left_parts, right_parts):
+        """
+        Concatenate the blocks of partitions with another set of blocks.
+
+        Parameters
+        ----------
+        axis : int
+            The axis to concatenate to.
+        left_parts : np.ndarray
+            NumPy array of partitions to concatenate with.
+        right_parts : np.ndarray or list
+            NumPy array of partitions to be concatenated.
+
+        Returns
+        -------
+        np.ndarray
+            A new NumPy array with concatenated partitions.
+
+        Notes
+        -----
+        Assumes that the `left_parts` and `right_parts` blocks are already the same
+        shape on the dimension (opposite `axis`) as the one being concatenated. A
+        ``ValueError`` will be thrown if this condition is not met.
+        """
+        result = super(PandasOnDaskDataframePartitionManager, cls).concat(
+            axis, left_parts, right_parts
+        )
+        if axis == 0:
+            return cls.rebalance_partitions(result)
+        else:
+            return result
+
+    @classmethod
+    def rebalance_partitions(cls, partitions):
```

Comment:
I think this method can be moved to the parent class.  Its logic is very complex and we shouldn't duplicate it.